src/api/app/models/review.rb

```diff
@@ -39,6 +40,8 @@ class NotFoundError < APIException
     end
   end
 
+  before_validation :set_entity_association
```

Comment:
Nope, the validation also runs for save etc. It basically runs always except you skip the validation or on some rare operations (e.g. touch). We even have a Rubocop rule that we don't skip model validations:
https://github.com/openSUSE/open-build-service/blob/master/.rubocop.yml#L265

I do it in before_validation so that I can rely on it in the validation, e.g.
``
if by_user && !entity
``
Otherwise I would need to query for the data twice (once in the validation and another time in the before_save to set the validation).

See also the ActiveRecordCallbacks here http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html